pkg/core/annotation.go

```diff
@@ -58,15 +61,28 @@ func (a Annotation) Format(s fmt.State, verb rune) {
 	}
 }
 
-func (a Annotation) Key() AnnotationKey {
+func (a *Annotation) Key() AnnotationKey {
 	return AnnotationKey{Capability: a.Capability.Name, ID: a.Capability.ID}
 }
 
+func (a *Annotation) Detach() {
+	a.isDetached = true
+	a.Node = nil
+}
+
+func (a *Annotation) IsDetached() bool {
+	return a.isDetached
+}
+
 func (m AnnotationMap) Update(other AnnotationMap) {
 	for k, v := range other {
 		if ex, ok := m[k]; ok {
 			// Update the contents not the pointer so existing annotation pointers are still valid
-			*ex = *v
+			newValue := *v
+			if ex.isDetached {
+				newValue.isDetached = true
```

Comment:
Should this also clear out the `Node` since it's invalid when detached (ie, call `newValue.Detach()`)